python/cudf/cudf/_lib/replace.pyx

```diff
@@ -137,14 +137,18 @@ def replace_nulls(
         raise ValueError("Cannot specify both 'value' and 'method'.")
 
     if method:
-        return replace_nulls_fill(input_col, method)
+        result = replace_nulls_fill(input_col, method)
     elif is_scalar(replacement):
-        return replace_nulls_scalar(
+        result = replace_nulls_scalar(
             input_col,
             as_device_scalar(replacement, dtype=dtype)
         )
     else:
-        return replace_nulls_column(input_col, replacement)
+        result = replace_nulls_column(input_col, replacement)
+
+    if is_decimal_dtype(result.dtype) and is_decimal_dtype(input_col.dtype):
+        result.dtype.precision = input_col.dtype.precision
```

Comment:
I think once https://github.com/rapidsai/cudf/pull/7788 lands, we can just use `_copy_type_metadata` here.

Reply:
Added a `depends on` to the PR description